src/node/internal/internal_timers_promises.ts

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) 2017-2022 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+//
+// Adapted from Node.js. Copyright Joyent, Inc. and other Node contributors.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to permit
+// persons to whom the Software is furnished to do so, subject to the
+// following conditions:
+//
+// The above copyright notice and this permission notice shall be included
+// in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+// OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN
+// NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+// OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE
+// USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+/* eslint-disable @typescript-eslint/require-await,@typescript-eslint/no-deprecated,prefer-spread */
+
+import * as timers from 'node-internal:internal_timers';
+
+export async function setTimeout(
+  ...args: Parameters<typeof timers.setTimeout>
+): Promise<ReturnType<typeof timers.setTimeout>> {
+  return timers.setTimeout.apply(timers, args);
```

Comment:
I fail to understand how this is a promise version?

Reply:
Maybe there should be tests :)

Reply:
The `timers/promises` API is quite different: https://nodejs.org/api/timers.html#timers-promises-api

Reply:
Maybe ask this guy https://github.com/jasnell/awaitable-timers ;)

(I found the repo when I was thinking about implementing it in unenv)

Reply:
Ah, yeah it seems I forgot certain features. This basically mimics promises behavior by surrounding it with async.

Reply:
Fixed. Thanks!